### PR TITLE
Profile Configuration, UI for Handling False Positive & Report Preferences 

### LINKED ISSUE #611: UI for false positive analysis & profile configuration
Need to implement UI for false positive analysis & profile configuration. 

In initial implementation we are going with only one profile for project. 

When the user uploads an app file, a default profile will be created along with project. This profile's id can be accessed using `project.active_profile_id`.

### 1. Replace API calls in current settings page with profile endpoints:
Changed endpoints are listed here, other endpoints remain the same for now (eg: manual scan)
- [x] update device preferences: `PUT /api/profiles/1/device_preference` with `{"platform_version": "0", "device_type": "0"}`
- [x] update API scan options: `PUT /api/profiles/1/api_scan_options` with `{"api_url_filters": ""}`
- [x] update unknown analysis status: `PUT /api/profiles/1/unknown_analysis_status` with `{"status": true}`
- [x] get Github integration: `GET /api/profiles/1/github_integration`
- [x] get Jira integration: `GET /api/profiles/1/jira_integration`

##### For the false positive, 
- [x] dropdown with "Mark as (passed|low|medium|high|critical)".
- [x] on change, show popup with 2 options `for the current file only` & `for all future analyses` 
- [x] After save, the risk value display should show a `user-edited` icon on the side.
- [x] Edit Overridden risk
API for overriding risk: 
```
PUT /api/files/<file_id>/vulnerability_preferences/<vuln_id>/risk
{
     "risk": <int>,
     "all": <bool>
}
```
API for reseting risk: 
```
DELETE /api/files/<file_id>/vulnerability_preferences/<vuln_id>/risk
{
     "all": <bool>
}
```
##### For ignoring analysis,
- [x] button with "Ignore this vulnerability".
- [x] on click, show popup with 2 options `for the current file only` & `for all future analyses` 
- [x] After save, ignored analysis will get removed from current analysis list and appear in a disabled list (*ignored analyses*)at the bottom of the page.

API for setting ignore: 
```
PUT /api/files/<file_id>/vulnerability_preferences/<vuln_id>/ignore
{
    "ignore": <bool>,
     "all": <bool>
}
```
where `all` `false` will save for current file, `true` will save for all analyses (i.e., will get saved to corresponding profile).

Once updated,
- updated risk will be available at `analysis.overriden_risk`
- ignored value will be available at `analysis.is_ignored`

### 3. Report customization:
- [x] Show report customization section in settings page. Currently only one option is there: checkbox with *Show ignored analyses*

<div id="vuln-pref-settings"></div>

### 4. Vulnerability preferences in settings page
- [x] Update profile's vulnerability preferences from settings
```
GET /api/profiles/<profile_id>/vulnerability_preferences/<vuln_id>
```
```
PUT /api/profiles/<profile_id>/vulnerability_preferences/<vuln_id>
{
    "ignore": <bool>,
    "risk": <int>
}
```
project_profile_vuln_pref_settings

---

Backend is WIP here: https://github.com/appknox/mycroft/pull/902

### PR DESCRIPTION
**Pull request by yashwin**
Analysis Settings(File)
- [x] Edit each analysis  
- [x] Override Analysis - Handle False Positive
- [x] Ignore/Hide Analysis
- [x] Edit Overridden Analysis
- [x]  Unhide Analysis
- [x] Show Ignored/Hidden analysis at the end - Blurred
- [x] Show User Edited Icon and show text on hover

Analysis Settings(Project)
- [x] Move analysis settings to another tab
- [x] Show all the analysis in the project
- [x] Override Analysis & Edit Overridden analysis
- [x]  Reset Overridden Analysis
- [x] Hide & Unhide Analysis

Report Customisation
- [x] Show Report Configuration 

Changes in API for settings with profile endpoints
- [x] Device Preferences
- [x] API Scan Options
- [x] Unknown Analysis Status
- [x] ~Github Integration~
- [x] ~Jira Integration~

Things to be fixed

- [x] Immediately add ignored/hidden analysis to the list of hidden analysis
- [x] Sort analysis after overriding | after sorting by impact
- [x] Select the default risk when editing the overridden analysis 
- [x]  Cancel while editing overridden analysis  
-  [x] Fix sort by risk after overriding the risk

fixes #611

### CODE REVIEW

File: app/components/vulnerability-list.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  
+  vulnerabilityPreferences: (function() {
+    return this.get("store").query('vulnerability-preference', {id: this.get("project.activeProfileId")});
```
Comment (cosmosgenius) [inline app/components/vulnerability-list.js]: This should again be queryRecord!

File: app/adapters/vulnerability-preference.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import DRFAdapter from './drf';
+import ENV from 'irene/config/environment';
+import DataAdapterMixin from 'ember-simple-auth/mixins/data-adapter-mixin';
+
+export default DRFAdapter.extend(DataAdapterMixin, {
+  host: ENV.host,
+  namespace: ENV.namespace,
+  addTrailingSlashes: false,
+  authorizer: 'authorizer:irene',
+  query: function query(store, type, q) {
```
[the remark this answers is not in the document]
Comment (yashwin) [reply app/adapters/vulnerability-preference.js]: Not needed

File: app/components/vulnerability-list.js (hunk 1 above)
Comment (yashwin) [reply app/components/vulnerability-list.js]: No, query will do the job

Comment (cosmosgenius) [reply app/components/vulnerability-list.js]: cool!

**Pull request merged**